cpp/daal/src/algorithms/pca/transform/oneapi/pca_transform_dense_default_batch_oneapi_impl.i

```diff
@@ -169,7 +186,10 @@ services::Status TransformKernelOneAPI<algorithmFPType, method>::allocateBuffer(
     const algorithmFPType zero = 0.0;
     returnBuffer               = ctx.allocate(TypeIds::id<algorithmFPType>(), bufferSize, status);
     DAAL_CHECK_STATUS_VAR(status);
+
+    DAAL_ASSERT_UNIVERSAL_BUFFER(returnBuffer, algorithmFPType, bufferSize);
```

Comment:
Correctness of allocation was checked at line 188. No need to check buffer. It's up to you.

Reply:
The allocation may be moved to another place in the future. Since it builds only in debug mode, I believe it's better to check the buffer.